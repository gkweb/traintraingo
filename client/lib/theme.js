import React, {Component} from 'react'

const themes = {
  light: {
    'name': 'light',
    'values': {
      'primaryBg': '#f7f7f7',
      'secondaryBg': '#ddd',
      'tertiaryBg': '#bbb',
      'inversePrimaryBg': '#f7f7f7',
      'primary': '#4c4c4c',
      'secondary': '#9a9a9a',
      'tertiary': '#f7f7f7',
      'inversePrimary': '#4c4c4c',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#4c4c4c',
      'logoSecondary': '#fff',
      'input': '#222',
      'inputBg': '#fff',
      'inputBorder': '#ccc',
      'inputIcon': 'rgb(76, 76, 76, .75)',
      'link': '#4495d4',
      'linkBg': '#f7f7f7',
      'linkHoverBg': '#fff'
    }
  },
  dark: {
    'name': 'dark',
    'values': {
      'primaryBg': '#444',
      'secondaryBg': '#333',
      'tertiaryBg': '#222',
      'inversePrimaryBg': '#fff',
      'primary': '#fff',
      'secondary': '#ddd',
      'tertiary': '#000',
      'inversePrimary': '#000',
      'highlightPrimary': '#EEDB57',
      'highlightSecondary': '#ffB86f',
      'highlightTertiary': '',
      'logoPrimaryBg': '#5EC64C',
      'logoSecondaryBg': '#105C26',
      'logoPrimary': '#fff',
      'logoSecondary': '#fff',
      'input': '#fff',
      'inputBg': '#222',
      'inputBorder': '#000',
      'inputIcon': 'rgba(255, 255, 255, .25)',
      'link': '#fff',
      'linkBg': '#333',
      'linkHoverBg': '#222'
    }
  }
}

const ThemeManagementContext = React.createContext({
    activeTheme: themes.light.name,
    theme: themes.light // default value
  }
)

export {
  themes,
  ThemeManagementContext
}
