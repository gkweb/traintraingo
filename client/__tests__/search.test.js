import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import { themes } from './../lib/theme'
import Search from '../components/search'
import 'jest-styled-components'

const emptyHandler = () => console.log('null')

test('Search icon dissapears when focussed', () => {
  const c = mount(
    <ThemeProvider theme={themes.light.values}>
      <Search onSubmit={emptyHandler} onChange={emptyHandler} />
    </ThemeProvider>
  )

  // Simulate focus and blur to check toggling of SVG
  const input = c.find('input[type="text"]')
  input.simulate('focus')

  expect(c.find('svg').exists()).toBe(false)

  input.simulate('blur')

  expect(c.find('svg').exists()).toBe(true)
})

test('Search Snapshot', () => {
  const c = shallow(<Search onSubmit={emptyHandler} onChange={emptyHandler} />)
  expect(c).toMatchSnapshot()
})
