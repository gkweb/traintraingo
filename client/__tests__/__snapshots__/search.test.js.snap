// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Snapshot 1`] = `
<styled.form
  onSubmit={[Function]}
>
  <styled.input
    autoComplete="off"
    name="search_field"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
  <Styled(Component) />
</styled.form>
`;
